lib/bap_image/llvm_binary_stubs.cpp

```diff
@@ -0,0 +1,170 @@
+#include <string>
+
+#include "llvm_binary_stubs.h"
+#include "llvm_binary_stubs.hpp"
+
+extern "C" {
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/fail.h>
+#include <caml/alloc.h>
+#include <caml/custom.h>
+#include <caml/bigarray.h>
+}
+
+void llvm_binary_fail [[ noreturn ]](const char* message) {
+    ::caml_failwith(message);
+}
+
+namespace impl {
+
+using namespace llvm;
+using namespace llvm::object;
+
+Binary* from_value (::value v) {
+    CAMLparam1(v);
+    return *reinterpret_cast<Binary**>(Data_custom_val(v));
```

Comment:
There still should be `CAMLreturn`. `CAMLparam` must always match with `CAMLreturn`